_pages/algorithm-development/reloading-models.md

```diff
@@ -1,106 +1,104 @@
+Click the **Publish** button in the Web IDE, or use the [algo.publish()](https://docs.algorithmia.com/?python#publish-an-algorithm) method (or the equivalent method for your language of choice if not Python) if using the Algorithmia's API to [manage your algorithm]({{site.url}}{{site.baseurl}}/algorithm-development/algorithm-management).
 
+### Option 2: Updating your model on the fly, without changing version numbers
 
-### Option 2: Updating your Model immediately, without changing Version Numbers
+If you'll be updating your model very often (e.g., via an automated process that retrains the model hourly from a live data feed), versioning your algorithm on every single model change may be impractical. In this case, you may want to add automatic model reloading into your algorithm. This requires writing some custom code.
 
-If you'll be updating your model very often (e.g. via an automated process that retrains the model hourly from a live data feed), versioning your Algorithm on every single model change can be impractical. In this case, you may want to add automatic model reloading into your Algorithm. This requires writing some custom code.
+For example, consider a simple scikit-learn algorithm that loads a model file from a hosted data collection, accepts some text as input, and returns a number.
 
-For example, consider a simple scikit-learn Algorithm which loads a model file from Hosted Data, accepts some text as input, and returns a number:
-
-{% highlight python %}
+```python
 import Algorithmia
 from sklearn.externals import joblib
 
 client = Algorithmia.client()
 
-modelFile = client.file('data://username/demo/mymodel.pkl').getFile().name
-model = joblib.load(modelFile)
+model_file = client.file('data://ALGO_OWNER/ALGO_NAME/MODEL_FILE.pkl').getFile().name
+model = joblib.load(model_file)
+
+def apply(input):
+    return int(model.predict(input))
+```
 
-def apply(text):
-    return int(model.predict(text))
-{% endhighlight %}
-  
-We need to alter this code so that instead of only loading the model file once on warm-up, it reloads the model periodically.
+Let's alter this code so that instead of only loading the model file when the algorithm instance is first spun up, it reloads the model periodically.
 
-We do so by adding a function to reload the model. We'll call this function once, in the outer scope, to make sure it gets loaded at warm-up:
+We can do this by explcitly adding a function `reload_model()`, and implementing some logic to determine how often the model is reloaded. First, to ensure that the model gets loaded when the algorithm first warms up, we'll call the `reload_model()` function once in the global scope.
 
-{% highlight python %}
+```python
 import Algorithmia
 from sklearn.externals import joblib
 
 client = Algorithmia.client()
 
 def reload_model():
```

Comment:
hmm am I missing a step? It looks like model isn't being defined in global state anywhere?
@lemonez 

It's not something necessary for typical models; however if there is an ability to "swap out" model files based off of the API request coming in from the user, than this does work and make sense.

Reply:
I'm using the `global` keyword in `reload_model()` to define the model in the global scope. That func is called at the bottom when the algorithm is warmed up, so the model will be defined in global scope before `apply()` is ever called.

Reply:
I don't quite understand your second comment as far as what "something necessary for typical models" is referring to. Are you referring to my code as written? Or some other functionality where you'd pass it a flag in the payload that would indicate which model to use?

Reply:
I'm going to merge this because this was a specific customer ask and would like to get into LTS, but please still comment

Reply:
Yeah you don't need the `global` keyword unless you plan to mutate global state from within a function, it can be very useful for when you plan to swap out what's in global state, but since loading typically only happens once - it can be something as simple as `model = load_model()` where you can then just reference `model` from within the rest of your algorithm code.

Reply:
Right, usually it only happens once but the whole point of this page WAS to show the reloading part (mutating global state). We just had a customer using this, so that's why I updated and used the `global` keyword 🌏 👍🏼